test/testenv/appframework_utils.go

```diff
@@ -12,3 +16,64 @@ func GenerateAppSourceSpec(appSourceName string, appSourceLocation string, appSo
 		AppSourceDefaultSpec: appSourceDefaultSpec,
 	}
 }
+
+// GetPodAppStatus Get the app install status and version number
+func GetPodAppStatus(deployment *Deployment, podName string, ns string, appname string) (string, string, error) {
+	output := GetPodAppInstallStatus(deployment, podName, ns, appname)
+	status := strings.Fields(output)[1]
+	version, err := GetPodInstalledAppVersion(deployment, podName, ns, appname)
+	return status, version, err
+
+}
+
+// GetPodInstalledAppVersion Get the version of the app installed on pod
+func GetPodInstalledAppVersion(deployment *Deployment, podName string, ns string, appname string) (string, error) {
+	path := "etc/apps"
+	if strings.Contains(podName, "-indexer-") {
+		path = "etc/slave-apps/_cluster/default"
+	}
+	filePath := fmt.Sprintf("/opt/splunk/%s/%s/default/app.conf", path, appname)
+
+	confline, err := GetConfLineFromPod(podName, filePath, ns, "Version", "", false)
+	if err != nil {
+		logf.Log.Error(err, "Failed to get Version from pod", "Pod Name", podName)
+		return "", err
+	}
+	version := strings.TrimSpace(strings.Split(confline, " ")[2])
+	return version, err
+
+}
+
+// GetPodAppInstallStatus Get the app install status
+func GetPodAppInstallStatus(deployment *Deployment, podName string, ns string, appname string) string {
+	stdin := fmt.Sprintf("/opt/splunk/bin/splunk display app '%s'", appname)
+	command := []string{"/bin/sh"}
+	stdout, stderr, err := deployment.PodExecCommand(podName, command, stdin, false)
+	if err != nil {
+		logf.Log.Error(err, "Failed to execute command on pod", "pod", podName, "command", command, "stdin", stdin)
+		return "Failed"
```

Comment:
Nit: Return error..so that the consumer can differentiate between failed to run the command vs app not present ? Doing via strings is the not most efficient way. To keep the code styling consistent we should return errors explicitly.